### PR TITLE
Jacob W Burke

### PR DESCRIPTION
**Pull request by jwburke256**

### CODE REVIEW

**Review by zschok3 — COMMENTED** [review]
Followed naming conventions, and labelled everything correctly.  Easy to read and no needlessly complex variable names.

**Review by zschok3 — COMMENTED** [review]
Good changes, way to clean up some bugs while also making this more readable.

File: brevets/flask_brevets.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+app.logger.setLevel(logging.DEBUG)
+
+##################################################
+################### API Callers ################## 
+##################################################
+
+API_ADDR = os.environ["API_ADDR"]
+API_PORT = os.environ["API_PORT"]
+API_URL = f"http://{API_ADDR}:{API_PORT}/api/"
+
```
Comment (amontg14) [inline brevets/flask_brevets.py:36]: im assuming the issue reading port numbers must be somewhere around here or in the docker file

Comment (amontg14) [reply brevets/flask_brevets.py:36]: *docker compose

File: brevets/flask_brevets.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    except:
+        return flask.jsonify(
+                result={}, 
+                status=0,
+                message="Something went wrong, couldn't fetch a brevet!")
+
+#############
+
+if __name__ == "__main__":
+    app.run(port=port_num, host="0.0.0.0")
```
Comment (amontg14) [inline brevets/flask_brevets.py:165]: this all looks good

File: api/flask_api.py [hunk 3]
```diff
@@ -2,25 +2,31 @@
...
-# api.add_resource(...)
+api.add_resource(BrevetResource, "/api/brevet/<id>")
+api.add_resource(BrevetsResource, "/api/brevets")
+
 
 if __name__ == "__main__":
     # Run flask app normally
     # Read DEBUG and PORT from environment variables.
-    pass
+    app.run(port=port_num, host="0.0.0.0")
```
Comment (amontg14) [inline api/flask_api.py:32]: this all looks good

File: api/resources/brevets.py [hunk 4]
```diff
@@ -27,3 +27,16 @@
...
+        json_object = Brevet.objects().to_json()
+        return Response(json_object, mimetype="application/json", status=200) 
+    
+    def post(self):
+        # Read the entire request body as a JSON
+        # This will fail if the request body is NOT a JSON.
+        input_json = request.json
+
+        result = Brevet(**input_json).save()
+        return {'_id': str(result.id)}, 200
```
Comment (amontg14) [inline api/resources/brevets.py:42]: looks good

File: api/resources/brevet.py [hunk 5]
```diff
@@ -27,3 +27,17 @@
...
+        return Response(brevet, mimetype="application/json", status=200)
+
+    def put(self, id):
+        input_json = request.json
+        Brevet.objects().get(id=id).update(**input_json)
+        return '', 200
+
+    def delete(self, id):
+        Brevet.objects().get(id=id).delete()
+        return '', 200
```
Comment (amontg14) [inline api/resources/brevet.py:43]: looks good

File: api/database/models.py [hunk 6]
```diff
@@ -19,4 +23,6 @@ class Brevet(Document):
 		start_time: MongoEngine datetime field, required
 		checkpoints: MongoEngine list field of Checkpoints, required
     """
-    pass
+    brevet_dist_km = FloatField(required = True)
+    begin_date = StringField(required=True)
+    items = EmbeddedDocumentListField(Checkpoint)
```
Comment (amontg14) [inline api/database/models.py:28]: looks good

**Review by amontg14 — COMMENTED** [review]
This program looks good, but i was unable to successfully run it. kept running into the same error where it was unable to read information from the docker_compose file. I'm assuming this is due to some sort of error in the flask_brevets file. Overall the code is clean, just a shame I couldn't get it to run

**Pull request closed without merging**